Slow game speed step

qqkookie opened the pull request:
Slow game speed . code clean up.

qqkookie commented:
I can not figure out how to make a pull request on existing pull open request, so I make new one.

Broxzier commented:
The other PR now says "unknown repository", did you delete your repo and create a new fork?

For future reference, we're using the gitflow workflow: https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow

- To set up another remote called `upstream` that points to the original repository:
`git remote add upstream https://github.com/OpenRCT2/OpenRCT2.git`
- To start working on a new feature, branch off develop
`git checkout develop`
`git branch new-feature && git checkout new-feature`
Or for short: `git checkout -b new-feature`
- Just make commits on the new branch as you would normally:
`git add/commit/push ...`
- To solve conflicts, rebase on latest version of upstream:
`git fetch upstream`
`git rebase upstream/develop`
You may need to solve some conflicts then. Ther eare some useful tools to make it easier.

qqkookie commented:
So should I close this pull request and start from brand new fork and branch? Shall I destroy the  2nd forked repo and make 3rd repo?

Broxzier commented:
You can continue commiting on develop, and after this gets merged it's better to reset it to be in line with upstream/develop. Unfortunately I don't think you can rename a branch after having created a pullrequest.

janisozaur commented:
> should I close this pull request and start from brand new fork and branch? Shall I destroy the 2nd forked repo and make 3rd repo?

Please don't. Just re-use one PR and update it accordingly.

qqkookie commented:
I don't know how to reuse existing, open  PR for updated commit. That is exact reason that I destroyed first forked repo and re-forked and re-issued the PR. I tied every buttons on the PC GitHub Desktop windows client, but I can not figure out the way to re-use open PR for updated commit.

Broxzier commented:
To update an existing PR, you only have to add commits to your `develop` branch, and push them to your remote repository here (Repository > Push).

Broxzier commented on src/openrct2/Game.h:
```diff
@@ -140,6 +140,13 @@ extern uint32_t gCurrentTicks;
 extern uint16_t gTicksSinceLastUpdate;
 extern uint8_t gGamePaused;
 extern int32_t gGameSpeed;
+enum GAMESPEED {
```
Please move this enum up to be with the other enums.

Broxzier commented on src/openrct2/GameState.cpp:
```diff
@@ -32,6 +32,9 @@
 using namespace OpenRCT2;
 
+// speed steps in percent. 25% is 1/4 of normal speed (100%).  
+static uint32_t GameSpeeds[GAMESPEED_HYPER+1] = { 0, 25, 100, 200, 400, 1000, 10000 };
```
Add a trailing comma after `10000` please

janisozaur replied:
Is this code really valid? The enum you use here has some oddities like not being contiguous and being assigned seemingly arbitrary values.

Broxzier commented on src/openrct2/Game.h:
```diff
@@ -140,6 +140,13 @@ extern uint32_t gCurrentTicks;
+    GAMESPEED_PAUSED = 0,
+    GAMESPEED_SLOW = 1,
+    GAMESPEED_NORMAL = 2,
+    GAMESPEED_TURBO = 5,
```
Why does it jump from 2 to 5?

Broxzier commented on src/openrct2/Context.cpp:
```diff
@@ -764,7 +764,7 @@ namespace OpenRCT2
             if (!gConfigGeneral.uncap_fps)
                 return false;
-            if (gGameSpeed > 4)
+             if (gGameSpeed > GAMESPEED_HYPER)
```
Accidental extra space

janisozaur commented:
Rather than merging, could you rebase this PR? It makes for a simpler history.

Also I wonder what toolchain are you using, as none of our CIs is able to compile the code you submitted?

qqkookie commented:
What is rebase? The Github client I am using (Windows  GitHub Desktop) has no menu or command  named "rebase".

qqkookie commented:
I am using Community edition of MS Visual Studio 2017. Here is information:

Microsoft Visual Studio Community 2017 
버전 15.7.5
VisualStudio.15.Release/15.7.5+27703.2042
Microsoft .NET Framework
버전 4.7.03056

설치된 버전: Community

Visual C++ 2017   00369-60000-00001-AA098
Microsoft Visual C++ 2017

ASP.NET and Web Tools 2017   15.0.40625.0
ASP.NET and Web Tools 2017

C# 도구   2.8.3-beta6-63029-08. Commit Hash: e9a3a6c0ba5b1fde8b1fff964bdfb3fb768ee2eb
IDE에 사용되는 C# 구성 요소입니다. 프로젝트 형식 및 설정에 따라 다른 버전의 컴파일러를 사용할 수 있습니다.

Favorite Documents   2.3.2
Favorites menu for Visual Studio IDE.

GitHub.VisualStudio   2.5.4.3349
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

JavaScript 언어 서비스   2.0
JavaScript 언어 서비스

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio VC 패키지   1.0
Microsoft Visual Studio VC 패키지

NuGet 패키지 관리자   4.6.0
Visual Studio의 NuGet 패키지 관리자. NuGet에 대한 자세한 내용은 http://docs.nuget.org/를 참조하십시오.

Open Command Line   2.1.212
Opens a command line at the root of the project. Support for all consoles such as CMD, PowerShell, Bash etc. Provides syntax highlighting, Intellisense and execution of .cmd and .bat files.

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Google Test용으로 작성된 단위 테스트와 함께 Visual Studio의 테스트 도구를 사용합니다. 사용 약관 및 타사 고지 사항은 확장 설치 디렉터리에서 확인할 수 있습니다.

Visual Basic 도구   2.8.3-beta6-63029-08. Commit Hash: e9a3a6c0ba5b1fde8b1fff964bdfb3fb768ee2eb
IDE에서 사용하는 Visual Basic 구성 요소입니다. 프로젝트 형식 및 설정에 따라 다른 버전의 컴파일러를 사용할 수 있습니다.

Visual Studio Code 디버그 어댑터 호스트 패키지   1.0
Visual Studio에서 Visual Studio Code 디버그 어댑터를 호스트하기 위한 Interop 계층

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Windows Machine Learning Generator Extension   1.0
Windows Machine Learning Visual Studio Extension Detailed Info

qqkookie commented:
Character encoding may be culprit. I am using UTF-8 as default encoding for my Visual Studio source code editor.  But no problem to compile without adding using /utf-8 flag. What Encoding should I have use?

qqkookie commented:
It seems that there is some race condition or build order dependency. The exactly same "rebuild" (no change between multiple rebuild tests) does succeed mostly, but randomly fails sometimes.   

The linker tries to use library "openrct2.lib" before the the on-going linking is finished,
There are some anomaly in parallel linkage of openrct2-win.vcxproj. The dependency should be fixed. 

=============================================
..............................
4>RideRatings.cpp
4>sawyercoding_test.cpp
4>gtest-all.cc
3>   C:\GitHub\OpenRCT2\bin\openrct2.lib 라이브러리 및 C:\GitHub\OpenRCT2\bin\openrct2.exp 개체를 생성하고 있습니다.
4>TestData.cpp
4>tests.cpp
4>StringTest.cpp
4>TileElements.cpp
3>openrct2-dll.vcxproj -> C:\GitHub\OpenRCT2\bin\openrct2.dll
5>------ 모두 다시 빌드 시작: 프로젝트: openrct2-win, 구성: Debug x64 ------
6>------ 모두 다시 빌드 시작: 프로젝트: openrct2-cli, 구성: Debug x64 ------
6>Cli.cpp
5>openrct2-win.cpp
5>LINK : fatal error LNK1104: 'openrct2.lib' 파일을 열 수 없습니다.
5>C:\APPS\MSVS-17CE\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(676,5): error MSB6006: "link.exe"이(가) 종료되었습니다(코드: 1104).
5>"openrct2-win.vcxproj" 프로젝트를 빌드했습니다. - 실패
4>tests.vcxproj -> C:\GitHub\OpenRCT2\bin\tests.exe
4>Copies test data to output directory.
4>C:\GitHub\OpenRCT2\test\tests\testdata\images\logo.png
4>C:\GitHub\OpenRCT2\test\tests\testdata\keys\Player-56f4afb74622a23bd2539ee701fe1b2c13d7e6ba.pubkey
4>C:\GitHub\OpenRCT2\test\tests\testdata\keys\Player.privkey
4>C:\GitHub\OpenRCT2\test\tests\testdata\parks\bpb.sv6
4>C:\GitHub\OpenRCT2\test\tests\testdata\parks\tile-element-tests.sv6
4>C:\GitHub\OpenRCT2\test\tests\testdata\ratings\bpb.sv6.txt
4>6 File(s) copied
6>openrct2-cli.vcxproj -> C:\GitHub\OpenRCT2\bin\openrct2-cli.exe
========== 모두 다시 빌드: 성공 5, 실패 1, 생략 0 ==========

Broxzier commented:
Remove the openrct2.dll file from the bin folder and build again.

I've explained how to perform a rebase already to you [here](https://github.com/OpenRCT2/OpenRCT2/pull/7816#issuecomment-407442146) and [here](https://github.com/OpenRCT2/OpenRCT2/pull/7822#issuecomment-407565188).

Broxzier commented:
Visual Studio has nice integration for merging conflicts. Open the Team Explorer window, click home > changes > conflicts. There's a list of files with conflicts, click on them and choose merge. Then in the new tab that opens, select the lines you want to add. The "source" is the commit you're rebasing, the "target" is upstream/develop. All changes made in the target should stay in, unless your chances made them obsolete.

`git rebase --skip` will remove the commit from history, you should not do that for commits you want to keep.

qqkookie commented:
I tied to files with  >>>>>>>>>>>>> 's. But the files are so badly mixed up with new version of code and old version(s), I can not reliably update the, to new version. I'd like to destroy  current repo and re-fork and copy new version source to the new repo and commit , and do new pull request again with only single commit.  I am very confused,

janisozaur commented:
> I'd like to destroy current repo and re-fork

Don't do that, it destroys history and creates unnecessary mess. Cleanup the code in your current fork and update the branch with this PR. There is no need to destroy the repository. It's fine to squash the commits if you want to.

qqkookie commented:
I still can not figure out what the "rebase" does. It seems that "rebase" is merging or replacing  multiple commits into single commit. So Why not just destroy repo and re-fork and do single commit with final  version once, instead of the "rebase". I can not understand rationale behind the need of rebase.

jensj12 commented:
It can be helpful to first squash all your own commits into one (with `rebase HEAD~4 -i`), that squashes out all intermediate versions, but removes history. When there's a conflict, the conflicted parts consists of two parts. One is what you made, the other is what is now on develop. The idea is that you move your own changes to the new develop version and then strip out your old edits, so only one of the blocks of code should remain.
If you really want to redo everything, perform a hard reset to OpenRCT2/develop so you can at least keep the same PR.
When you are done rebasing (or redoing everything), make sure you force push (`git push -f`) without pulling first.

Broxzier commented:
Huh... I did the rebase and when I pushed the changes this PR got closed. The reopen button is disabled.

Broxzier commented:
```
$ git push qqkookie qqkookie-develop:develop -f
fatal: HttpRequestException encountered.
   Er is een fout opgetreden bij het verzenden van het aanvraag.
Username for 'https://github.com': Broxzier
Counting objects: 36, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (36/36), done.
Writing objects: 100% (36/36), 4.07 KiB | 1.02 MiB/s, done.
Total 36 (delta 32), reused 0 (delta 0)
remote: Resolving deltas: 100% (32/32), completed with 22 local objects.
To https://github.com/qqkookie/OpenRCT2.git
 ! [remote rejected]     qqkookie-develop -> develop (permission denied)
error: failed to push some refs to 'https://github.com/qqkookie/OpenRCT2.git'
```
My local branch is called `qqkookie-develop` because develop already exists. I try to push to qqkookie's develop branch, but now I'm getting "permission denied". @janisozaur Any idea what I'm doing wrong?

janisozaur commented:
I can imagine two scenarios:
1. This PR is closed, so GitHub doesn't recognise it as a valid target for you to push.
2. Although opt-out, you _can_ prevent contributors of the upstream from pushing to your fork, there is a checkbox for that.

The reopen button is inactive and indeed the fork's `develop` seem to equal our `develop`, so we can only wait for @qqkookie to address it on his(?) end.

qqkookie commented:
Why this PR is closed?   I removed the most intermediate commits and rebased. Rebase was succesful.  Bu it left many conflits in source. Still the source code is mass. There are massive amount of conflicts  conflicts that I did not modified, (Mostly related to Russian unicode stuff, from new commits I guess ) and I can not tell which part is correct one.  There are so many conflicts to resolve ( most are not mine)

qqkookie commented:
I checked out again nd the source code is messy with many Russian conflicts. How can I revert back to old version without such Russian Unicode related stuff?

Broxzier commented:
First make sure you have your remotes set up correctly:
```
$ git remote add upstream https://github.com/OpenRCT2/OpenRCT2.git
```
Check if your output matches this:
```
$ git remote -v
origin  https://github.com/qqkookie/OpenRCT2.git (fetch)
origin  https://github.com/qqkookie/OpenRCT2.git (push)
upstream        https://github.com/OpenRCT2/OpenRCT2.git (fetch)
upstream        https://github.com/OpenRCT2/OpenRCT2.git (push)
```
Then fetch upstream and rebase on top of it:
```
$ git fetch upstream
From https://github.com/OpenRCT2/OpenRCT2
 * [new branch]          develop               -> upstream/develop
 * [new branch]          epic/plug-in          -> upstream/epic/plug-in
 * [new branch]          gif-recording         -> upstream/gif-recording
 * [new branch]          master                -> upstream/master
 * [new branch]          twitch-window         -> upstream/twitch-window
 * [new branch]          vsts                  -> upstream/vsts
 * [new branch]          windows/game-bar-demo -> upstream/windows/game-bar-demo

$ git rebase upstream/develop -i
```
A text editor will open, remove all commits except for your own.

Then when conflicts occur DONT just add the files with conflicts and all, but solve them. Otherwise you'll have the same mess again. If you need help with that, you can ask for it here or on the gitter chat.

qqkookie commented:
I manually recovered the lost source file changes from the rebasing. Two times.  Committing some source changes and doing "rebase" seems to "delete" permanently the last commit I made .  So I had to repeat the same manual recovery of the lost changes by manually editing the source file, TWO times, actually 3 times including first time I made the changes. This time, I did not do the rebase, because rebase will wipe the recovered changes I made 3rd times. So What should I do now? The changed source is in "slow" branch. I fear that, if  I commit the new changes to the "develop" branch, they will be lost on next check out,  I assumed hat committing and check out should work as expected. But for some reason, such common sense expectation (the case of CVS system) seems not apply here in Git.  I lost 2 times the changes I made recently. I fear 3rd loss.

Broxzier commented:
git doesn't remove commits when rebasing. You must have done something like not marking it with `pick` or using `git rebase --skip` while in the progress. In the worst case where something does go missing unintentionally, you can always go back to the state you were in before the rebase with `git reflog`.

In the future, don't work on develop directly, that would make cleaning up this mess a lot easier. I've described how to do that in the first comment: https://github.com/OpenRCT2/OpenRCT2/pull/7816#issuecomment-407442146

qqkookie commented:
OK. next time, I will follow your advice of not working on 'develop'. I promise. But for now, I am at loss and don't know what to do next. I'd like to clean this mess from my ignorance, mistake, inexperience  and  want to recover sanity in current situation. The result of "git reflog ." is like this:

$ git reflog
e25b0336c (HEAD -> slow, origin/slow) HEAD@{0}: commit: Recovered the changs lost due to the "rebasing" procedure. lost two times. This is 2nd time.
e1a7fbc7e HEAD@{1}: checkout: moving from e1a7fbc7e6f1debeaf56c285beb39f3c168d4126 to slow
e1a7fbc7e HEAD@{2}: checkout: moving from develop to e1a7fbc7e6f1debeaf56c285beb39f3c168d4126
43a9228b8 (origin/develop, origin/HEAD, develop) HEAD@{3}: checkout: moving from e1a7fbc7e6f1debeaf56c285beb39f3c168d4126 to develop
e1a7fbc7e HEAD@{4}: checkout: moving from develop to e1a7fbc7e6f1debeaf56c285beb39f3c168d4126
43a9228b8 (origin/develop, origin/HEAD, develop) HEAD@{5}: checkout: moving from master to develop
0aff800d4 (tag: v0.2.0, upstream/master, origin/master) HEAD@{6}: checkout: moving from develop to master
43a9228b8 (origin/develop, origin/HEAD, develop) HEAD@{7}: checkout: moving from 9c2f20e5838cda9d183b69053358b1cb4cc7e7a0 to develop
9c2f20e58 HEAD@{8}: commit: Recovering all changes that were lost during rebasing. (07-25)
3cb79df11 HEAD@{9}: commit: Slow game speed
80f67247d (upstream/develop) HEAD@{10}: rebase -i (start): checkout upstream/develop
e2ec83bfb HEAD@{11}: reset: moving to HEAD~
e1a7fbc7e HEAD@{12}: checkout: moving from develop to e1a7fbc7e6f1debeaf56c285beb39f3c168d4126
43a9228b8 (origin/develop, origin/HEAD, develop) HEAD@{13}: commit (merge): Merge branch 'develop' of https://github.com/qqkookie/OpenRCT2 into develop
e1a7fbc7e HEAD@{14}: rebase -i (finish): returning to refs/heads/develop
e1a7fbc7e HEAD@{15}: rebase -i (continue): name consistency
e2ec83bfb HEAD@{16}: rebase -i (pick): Slow game speed
d4b8cbd34 HEAD@{17}: rebase -i (pick): Fix #4039: Add sprite font glyph for German opening quotes (#7811)
d64836a83 HEAD@{18}: rebase -i (pick): Add cyrillic banner font characters
f04bc921e HEAD@{19}: rebase -i (pick): Amend changelog
25164becb HEAD@{20}: rebase -i (pick): Add sprite glyphs for Cyrillic/Russian
37046291e HEAD@{21}: rebase -i (pick): Add trailing commas to some window source files
8770ee872 HEAD@{22}: rebase -i (pick): Apply clang-format changes to tests
d63252e46 HEAD@{23}: rebase -i (pick): Indent preprocessor directives
18485e3f3 HEAD@{24}: rebase -i (pick): Binpack function arguments together
126e8fd40 HEAD@{25}: rebase -i (pick): Enforce not breaking before assignments and function names
8250cd02d HEAD@{26}: rebase -i (pick): Apply format on new/updated code before rebase
745b9980c HEAD@{27}: rebase -i (pick): Correct formatting for files in the project root
a2dd33d7b HEAD@{28}: rebase -i (pick): Correct formatting in remaining files in libopenrct2
a74779032 HEAD@{29}: rebase -i (pick): Correct formatting for source files up to the rct2 folder

Now, What should I do?
Please tell me exact action to do next.

qqkookie commented:
Is anybody to help me? What should I do?

qqkookie commented:
I'd like to reset the current 'develop' branch. I have no prior experince of rest. And what is real result after the reset? Just empty repo or old state repo just after initial fork few days ago or same as freshly  re-forked state today. I wish that this repo state become same as just forked one today.

Broxzier commented:
By resetting, you'll loose the commits that you've made for this PR. You could rename this branch to make a backup, then checkout a new develop branch that points to upstream/develop.

```
git branch -m new-branch-with-this-history
git branch develop --set-upstream-to=upstream/develop
```
Then you'll have a develop branch that matches upstream/develop, though it'll break this PR.

qqkookie commented:
So  the exact line I will type into  my Git console:
git reset
git branch -m new-branch-with-this-history
git branch develop --set-upstream-to=upstream/develop

What the 'new-branch-with-this-history'? Is it option with long name? It seems strange for command argument. Can I copy and paste three three lines exactly character by character as it appears as above?

Broxzier commented:
No need to do `git reset`. And the argument after `-m` is the new branch name. Assuming you're on the develop branch without any local changes, just copy the two lines I posted if you want to reset develop. Your changes will still be available in the new branch then.

I've rebased your branch on a branch of mine, I can try again to push it here, but the last time it closed the PR: https://github.com/Broxzier/OpenRCT2/commits/qqkookie-develop

qqkookie commented:
Cookie@KOOK-PC MINGW64 /c/GitHub/OpenRCT2 (slow)
$ git checkout
Your branch is up to date with 'origin/slow'.

Cookie@KOOK-PC MINGW64 /c/GitHub/OpenRCT2 (slow)
$ git checkout develop
Switched to branch 'develop'
Your branch is up to date with 'origin/develop'.

Cookie@KOOK-PC MINGW64 /c/GitHub/OpenRCT2 (develop)
$ git branch -m old-history

Cookie@KOOK-PC MINGW64 /c/GitHub/OpenRCT2 (old-history)
$ git branch develop --set-upstream-to=upstream/develop
fatal: branch 'develop' does not exist

Cookie@KOOK-PC MINGW64 /c/GitHub/OpenRCT2 (old-history)

Git say that "fatal: branch 'develop' does not exist"
It seems just renamed develop branch to old-history

Broxzier commented:
Then try:

```
git checkout -b develop
git reset upstream/develop --hard
git branch --set-upstream-to=upstream/develop
```

qqkookie commented:
OK. Done. Now I will create the new "slow-game" branch as you recommended.  And I will put all my source changes to the the newly created "slow-game" branch. Am I doing right?  Then how can I change the branch related to this PR to new "qqkookie:slow-game branch", not current 'qqkookie:develop' branch ?

qqkookie commented:
Hmm. then I will do steps in orders as list below:

Step 1. Currently I am in 'develop' branch with no change, no edit and no commit..
Step 2. I will create new "slow-game" branch : "git  -b slow-game"
Step 3. I will switch to "slow-game" branch : "git checkout slow-game"
Step 4. Then I will add and recover the lost source changes and edits to the 'slow-game' branch by manually editing sources of the recently checkout 'slow-game' branch.
Step 5. Then commit the edits on 'slow-game' branch by : " git commit"
Step 6. then I will switch back to develop branch by : "git checkout develop"
Step 7.  I will do the dreaded "rebase" by "git fetch upstream" and "git rebase upstream/develop"
Step 8. If there are any conflicts, I will resolve the conflicts by manually editing the 'develop' branch.
Step 9. Then I will commit the conflict-resolved sources to the 'develop' branch by : "git commit develop"

Am I doing right? 
It doubt that there must be some step to transfer the changes I made on "slow-game" branch, from 'slow-game' branch to 'develop' branch,  between Step 5  and Step 7 above.

Broxzier commented:
Step 2: It's `git branch <name>`
step 4-5: should be done on develop instead, so this PR gets updated...
Step 9: no... you solve conflicts while rebasing, not afterwards. Otherwise you're left with all these conflict markers in the code like now.

Please check this checkbox on the right, so I can push the rebased changes here:
![firefox_2018-07-26_21-38-43](https://user-images.githubusercontent.com/9705046/43286698-263b6b80-9123-11e8-99db-e2cbc6811b87.png)
Then you need can improve this further by working on the develop branch.

qqkookie commented:
Thank you, Then I will edit the develop branch. 
And The "Allow edits from maintainer" is checked state already. I did't changed it. It was checked from the beginning.  I invited you as collaborator. although I don't expect it will help to mitigate the situation, 
The feature setting on this repo  is  "Restrict editing to collaborators only (Public wikis will still be readable by everyone.)" is "UNCHECKED". Should I check it?

Lastorder-DC commented:
@qqkookie That setting is for repo's wiki, not repo itself.

qqkookie commented:
Hmmm I can not commit. I added the new changes on the local (WindowPC) copy of develop branch. And did issued " git commit" command, but it failed.

Cookie@KOOK-PC MINGW64 /c/GitHub/OpenRCT2 (develop)
$ git commit
On branch develop
Your branch is up to date with 'upstream/develop'.

Changes not staged for commit:
        modified:   contributors.md
        modified:   data/language/en-GB.txt
        modified:   distribution/changelog.txt
        modified:   src/openrct2-ui/title/TitleSequencePlayer.cpp
        modified:   src/openrct2-ui/windows/TopToolbar.cpp
        modified:   src/openrct2/Context.cpp
        modified:   src/openrct2/Game.h
        modified:   src/openrct2/GameState.cpp
        modified:   src/openrct2/localisation/Localisation.cpp
        modified:   src/openrct2/localisation/Localisation.h
        modified:   src/openrct2/localisation/StringIds.h
        modified:   src/openrct2/title/TitleScreen.cpp

no changes added to commit

Cookie@KOOK-PC MINGW64 /c/GitHub/OpenRCT2 (develop)

qqkookie commented:
I am running out my patience on this PR. I would like to restart all over again from the scratch on  the new, clean, blank sheet. I wish to close this PR and destroy current PR and re-fork again. And start all over again from the new fork and new PR. May I do so?

jensj12 commented:
Looks like you tried to push to OpenRCT2/develop instead of your own fork, which seems to be lost completely. I don’t know a lot about git but this seems to be strictly worse than what I once reset my repo for (but that somehow allowed me to keep the same PR).

qqkookie commented:
Thank you for all the help by Broxzier and others. But I am so lost that it is way better to start again than trying to fix hopelessly broken current situation.
